LogicalCleaning/Logical_Data_Cleaning.R

```diff
@@ -0,0 +1,51 @@
+### Logical Data Cleaning with cleanR package
+rm(list = ls())
+today <- Sys.Date()
+
+## Step 1: load required packages
+library(tidyverse)
+library(openxlsx)
+# install.packages("devtools")
+# remotes::install_github("axmedmaxamuud/cleanR")
+library(cleanR)
+
+## Step 2: load data & create blank cleaning log book file
+raw_data <- cleanR::data
+my_logbook <- cleanR::logbook() # this will be blank logbook that you can use to export at the end.
+
+## Step 3: Start running your data checking 
+# check survey duration
+check_form_time <- survey_time(df = raw_data, time_min = 20, time_max = 40) %>% 
+  filter(CHECK_interview_duration %in% c("Too long", "Too short")) %>% 
+  log_sheet(question.name = "interview_duration", 
+            issue = "survey filled with less/long time, please check",
+            action = "check")
+# merge check_form_time with your logbook using the rbind function
+my_logbook <- rbind(my_logbook, check_form_time)
+
+# check other responses 
+# first make a list of all other columns included in your data
+other_columns <- c("camp_structure_other")
+
+check_others <- check_other_responses(data = raw_data, other_columns = other_columns)
+
+# include other checks dataframe into your logbook
+my_logbook <- rbind(my_logbook, check_others)
+
+# check the age of respondents
+check_ki_age <- raw_data %>% 
+  filter(ki_age < 18) %>% 
```

Comment:
ki_age variable is RESPAge in Survey Designer. Should be amended accordingly.

Reply:
variable naming is aligned with the survey designer.